src/internal/pachsql/db.go

```diff
@@ -58,6 +70,102 @@ func OpenURL(u URL, password string) (*DB, error) {
 	return res, errors.EnsureStack(err)
 }
 
+// ListTables returns an array of SchemaTable structs that represent the tables.
+func ListTables(ctx context.Context, db *DB) ([]SchemaTable, error) {
+	query := `
+		SELECT schemaname, tablename
+		FROM pg_catalog.pg_tables
+		WHERE schemaname != 'pg_catalog'
+		  AND schemaname != 'information_schema'
+		ORDER BY schemaname, tablename;
+	`
+	rows, err := db.QueryContext(ctx, query)
+	if err != nil {
+		return nil, errors.EnsureStack(fmt.Errorf("could not list tables: %s", err.Error()))
+	}
+	defer rows.Close()
+	var tables []SchemaTable
+	if err := sqlx.StructScan(rows, &tables); err != nil {
+		return nil, errors.EnsureStack(err)
+	}
+	return tables, nil
+}
+
+// GetScannedRows queries db and calls MapScanRows on the response rows.
+func GetScannedRows(ctx context.Context, db *DB, query string) ([]RowMap, error) {
+	rows, err := db.QueryContext(ctx, query)
+	if err != nil {
+		return nil, errors.EnsureStack(err)
+	}
+	defer rows.Close()
+	scannedRows, err := MapScanRows(rows)
+	if err != nil {
+		return nil, errors.EnsureStack(err)
+	}
+	return scannedRows, nil
+}
+
+// MapScanRows iterates through rows and returns an array of sqlx.MapScan results. This is useful
+// for when you want to capture the results, but don't know what columns the query can return
+// or when you don't want to create a struct to represent the columns.
+func MapScanRows(rows *sql.Rows) ([]RowMap, error) {
+	var processedRows []RowMap
+	for rows.Next() {
+		row := RowMap{}
+		if err := sqlx.MapScan(rows, row); err != nil {
+			return nil, errors.EnsureStack(err)
+		}
+		processedRows = append(processedRows, row)
+	}
+	return processedRows, nil
+}
+
+// GetMainColumn returns the 'key' column if it exists for SchemaTable table. Otherwise, it queries the db and returns
+// the column_name where ordinal_position is 1. This column can be used to order future queries against this table.
+func GetMainColumn(ctx context.Context, db *DB, table *SchemaTable) (string, error) {
+	keyExists, err := KeyColumnExists(ctx, db, table)
+	if err != nil {
+		return "", err
+	}
+	if keyExists {
+		return "key", nil
+	}
+	query := fmt.Sprintf(`
+		SELECT column_name FROM INFORMATION_SCHEMA.COLUMNS
+		WHERE table_schema = '%s'
+		  AND table_name = '%s'
+		  AND ordinal_position = 1;
```

Comment:
is Ordinal position guaranteed to give us the Primary key here?

Reply:
You're right that it doesn't work in general. My solution was to search for a 'key' column and if it didn't exist, then to default to whatever is in ordinal_postion = 1. That seems to work fine for the set of tables we have currently

Reply:
We ended up remove these